OpenDreamShared/Compiler/DMPreprocessor/DMPreprocessorLexer.cs

```diff
@@ -56,7 +56,15 @@ class DMPreprocessorLexer : TextLexer {
                             while (true) {
                                 bool isStar = GetCurrent() == '*';
 
-                                if (isStar && Advance() == '/') break;
+                                if (isStar && Advance() == '/')
+                                {
+                                    Advance();
+                                    while (GetCurrent() == ' ' || GetCurrent() == '\t')
+                                    {
+                                        Advance();
+                                    }
+                                    if(!AtEndOfSource) break;
```

Comment:
Wouldn't we want to always break, regardless of if we're at the end of the source?

Reply:
Wouldn't that prevent the exception from throwing because the `else if(AtEndOfSource)` doesn't fire?

Reply:
At this point the comment already ended, so it shouldn't be throwing.

Reply:
An excellent point.